### PR TITLE
ttl, variables: add system variables for ttl job schedule

### LINKED ISSUE #39725: Add system variables for ttl job scheduling

### PR DESCRIPTION
**Pull request by YangKeao**
### What problem does this PR solve?

Issue Number: close #39725 

Problem Summary:

Add the following system variable for ttl job schedule:

- tidb_ttl_job_run_interval
- tidb_ttl_job_schedule_window_start_time
- tidb_ttl_job_schedule_window_end_time
- tidb_ttl_scan_worker_count
- tidb_ttl_delete_worker_count

It also removes the `TTLFeatureGate`

### What is changed and how it works?

These global variables (except worker counts) are all tracked by an atomic variable, and set/get through the hook. The scan/delete worker count are set/get through scaling the workers in the ttl job manager.

This PR is based on https://github.com/pingcap/tidb/pull/39472

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

```release-note
Add system variables `tidb_ttl_job_run_interval`, `tidb_ttl_job_schedule_window_start_time`, `tidb_ttl_job_schedule_window_end_time`, `tidb_ttl_scan_worker_count`, `tidb_ttl_delete_worker_count` to control the ttl job management
```

### CODE REVIEW

**Comment by YangKeao** [comment]
PTAL @lcwangchao and @bb7133

**Review by lcwangchao — APPROVED** [review]

File: ttl/cache/infoschema.go [hunk 1]
```diff
@@ -40,7 +41,7 @@ func NewInfoSchemaCache(updateInterval time.Duration) *InfoSchemaCache {
 
 // Update updates the info schema cache
 func (isc *InfoSchemaCache) Update(se session.Session) error {
-	is := se.SessionInfoSchema()
+	is := se.GetDomainInfoSchema().(infoschema.InfoSchema)
```
Comment (YangKeao) [inline ttl/cache/infoschema.go:44]: Also submitted a minor optimization. Thanks for the suggestion from @lcwangchao

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**